docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,107 @@
+- Grace Period: The length of time to give suspended SNOs to diagnose and fix issues causing downtime. Afterwards, they will have one tracking period to demonstrate an acceptable amount of downtime before disqualification.
+
+When the [downtime tracking system](storage-node-downtime-tracking.md) adds an entry to the nodes_offline_time table, if the node is not already suspended, we check the total amount of downtime for the tracking period, e.g. if the tracking period is 30 days, we sum downtime for the last 30 days. If the total is greater than the allowed downtime, the node is suspended for a length of time equal to the grace period + one tracking period.
+
+Suspended nodes are not selected for uploads, and all pieces they hold are considered unhealthy. 
+
+### Disqualifying or reinstating suspended nodes
+
+In order to be reinstated, suspended nodes need to prove to us that they have changed their behavior. This requires that we observe a period of acceptable downtime from them. Otherwise, they are disqualified.
+
+For this purpose we implement a new chore for evaluating suspended nodes. The evaluation chore runs on a configurable interval, and checks to see if any downtime-suspended nodes have reached the end of their suspension (grace period + tracking period). For these nodes, check the total downtime for the tracking period. If a node's downtime has fallen within the allowed downtime range, it is reinstated. Otherwise, it is disqualified.
+
+## Rationale
+
+### Revert to using audits to track downtime?
+This blueprint for disqualification was written to work with the currently implemented downtime tracking system. However, questions have recently been raised as to whether the change from using audits to check downtime to the current 'downtime tracking' system was necessary. The impetus for the transition was that a node could become disqualified by being offline for only a short period of time due to the chance that it could be audited many times in quick succession. That said, auditing does give us some indication of node downtime. Is implementing a new system for tracking downtime worth the additional resources if the audit system is already randomly reaching out to nodes? Ultimately what we care about is whether nodes are available to receive and deliver data and verifying that said data is intact. If a node is offline when audited, neither of these cases are true. Whether the node was offline for five minutes or five hours, it still failed to present the data when requested. However, it would be remiss of us to simply ignore the plight of the unlucky node. A certain amount of downtime is expected, and when, by chance, nodes can be disqualified within the expected window, it seems that time indeed is a factor which should be taken into account. Is it possible for us to salvage the previous system and account for its flaws? Perhaps. Here is a quick idea for discussion, the details of which would require further investigation:
```

Comment:
> Ultimately what we care about is whether nodes are available to receive and deliver data and verifying that said data is intact.

That is not hitting the target. Lets say a node is offline. After a few hours the repair service will step in and repair it. -> We don't care if the data is intact because our system is handing that already. The magic question here is a different one. Can we expect the node to come back online or will it stay offline? If the node comes back online do we want to give it a new chance? If the node is going on and off frequently that will generate unnecessary repair cost on our side. The magic question is do we want to keep working with such a storage node. If it generates us too much unnecessary repair cost we should DQ it.